### PR TITLE
Ca refactor/user model and others

### PR DESCRIPTION
**Pull request by cagodoy**
🐒

### CODE REVIEW

**Comment by cagodoy** [comment]
@mtavano

File: src/app/api/trip/trip.model.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        next(err)
+      } else if (doc != null) {
+        next(new Error(`${query} combination already exist`))
+      }
+
+      next()
+    })
+  })
+
+tripSchema.method('validater', function () {
```
Comment (mtavano) [inline src/app/api/trip/trip.model.ts]: we must to delete this :trollface:

File: src/app/api/user/user.model.ts [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import * as crypto from 'crypto'
 import * as mongoose from 'mongoose'
+import {ITERATIONS, KEY_LENGTH, ALGORITHM} from './user.constants'
```
Comment (mtavano) [inline src/app/api/user/user.model.ts]: rata

File: src/app/api/user/user.model.ts [hunk 3]
```diff
@@ -24,6 +25,17 @@ let userSchema = new mongoose.Schema({
...
+  paymentWay: {
+    type: String
+  },
+  theRabbit: {
+    type: [mongoose.Schema.Types.ObjectId],
+    ref: 'Deck'
+  },
+  cash: {
+    type: Number,
+    defaut: 0
```
Comment (mtavano) [inline src/app/api/user/user.model.ts]: `defaut` ?? change to `default` monkey! 🐒

**Comment by mtavano** [comment]
🚀  skt1

**Pull request merged**